DiceKit/MinimizationExpressionResult.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  MinimizationExpressionResult.swift
+//  DiceKit
+//
+//  Created by Logan Johnson on 8/15/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import Foundation
+
+public struct MinimizationExpressionResult<MinimizationExpressionResult: protocol<ExpressionResultType, Equatable>>: Equatable {
+    
+    public let minimizationExpressionResult: Int
+    
+    public init(_ minimizationExpressionResult: Int) {
+        self.minimizationExpressionResult = minimizationExpressionResult
+    }
+    
+}
+
+// MARK: - CustomStringConvertible
+
+extension MinimizationExpressionResult: CustomStringConvertible {
+    
+    public var description: String {
+        return "\(minimizationExpressionResult)"
+    }
+    
+}
+
+// MARK: - CustomDebugStringConvertible
+
+extension MinimizationExpressionResult: CustomDebugStringConvertible {
+    
+    public var debugDescription: String {
+        return "\(String(reflecting: minimizationExpressionResult))"
+    }
+    
+}
+
+// MARK: - Equatable
+
+public func == <L>(lhs: MinimizationExpressionResult<L>, rhs: MinimizationExpressionResult<L>) -> Bool {
+    return Int(lhs.minimizationExpressionResult.value) == Int(rhs.minimizationExpressionResult.value)
```

Comment:
These casts are a workaround but need to be removed, what should the type be throughout these result structs? I've tried a few different things but they always seem to break their protocols.